File: frontend/src/api/sla.ts
/**
 * SLA policy API calls.
 */
import apiClient from './client';

export interface SLAPolicy {
  id: string;
  name: string;
  user_id: string | null;
  storage_id: string;
  expected_frequency_hours: number;
  alert_webhook: string | null;
  alert_email: string | null;
  is_active: boolean;
}

export interface SLAPolicyListResponse {
  policies: SLAPolicy[];
  total: number;
}

export interface CreateSLAPolicyRequest {
  name: string;
  storage_id: string;
  user_id?: string | null;
  expected_frequency_hours: number;
  alert_webhook?: string | null;
  alert_email?: string | null;
  is_active?: boolean;
}

export interface UpdateSLAPolicyRequest {
  name?: string;
  expected_frequency_hours?: number;
  alert_webhook?: string | null;
  alert_email?: string | null;
  is_active?: boolean;
}

export async function listSLAPolicies(): Promise<SLAPolicyListResponse> {
  const response = await apiClient.get<SLAPolicyListResponse>('/sla/policies');
  return response.data;
}

export async function createSLAPolicy(data: CreateSLAPolicyRequest): Promise<SLAPolicy> {
  const response = await apiClient.post<SLAPolicy>('/sla/policies', data);
  return response.data;
}

export async function updateSLAPolicy(id: string, data: UpdateSLAPolicyRequest): Promise<SLAPolicy> {
  const response = await apiClient.put<SLAPolicy>(`/sla/policies/${id}`, data);
  return response.data;
}

export async function deleteSLAPolicy(id: string): Promise<void> {
  await apiClient.delete(`/sla/policies/${id}`);
}

File: backend/app/schemas/sla.py
"""SLA policy schemas."""

from pydantic import BaseModel, Field


class SLAPolicyCreate(BaseModel):
    name: str = Field(..., min_length=1, max_length=200)
    user_id: str | None = None
    storage_id: str
    expected_frequency_hours: int = Field(default=24, ge=1, le=8760)
    alert_webhook: str | None = Field(default=None, max_length=500)
    alert_email: str | None = Field(default=None, max_length=255)
    is_active: bool = True


class SLAPolicyUpdate(BaseModel):
    name: str | None = Field(default=None, max_length=200)
    expected_frequency_hours: int | None = Field(default=None, ge=1, le=8760)
    alert_webhook: str | None = None
    alert_email: str | None = None
    is_active: bool | None = None


class SLAPolicyResponse(BaseModel):
    id: str
    name: str
    user_id: str | None
    storage_id: str
    expected_frequency_hours: int
    alert_webhook: str | None
    alert_email: str | None
    is_active: bool

    class Config:
        from_attributes = True


class SLAPolicyListResponse(BaseModel):
    policies: list[SLAPolicyResponse]
    total: int
